src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -504,33 +506,18 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
                                 "Challenge header is empty");
                     }
                 } else {
-
                     // AAD server returns 401 response for wrong request
                     // messages
                     Logger.v(TAG, "401 http status code is returned without authorization header");
                 }
             }
 
-            if (response.getBody() != null) {
-
-                // Protocol related errors will read the error stream and report
-                // the error and error description
-                Logger.v(TAG, "Token request does not have exception");
-                result = processTokenResponse(response);
-                ClientMetrics.INSTANCE.setLastError(null);
-            }
-
+            // Protocol related errors will read the error stream and report
+            // the error and error description
+            result = processTokenResponse(response);
+            
             if (result == null) {
-                // non-protocol related error
-                String errMessage = null;
-                byte[] message = response.getBody();
-                if (message != null) {
-                    errMessage = new String(message);
-                } else {
-                    errMessage = "Status code:" + String.valueOf(response.getStatusCode());
-                }
-
-                Logger.v(TAG, "Server error message:" + errMessage);
+                ClientMetrics.INSTANCE.setLastError(null);
                 if (response.getResponseException() != null) {
```

Comment:
With current implementation we're swallowing this exception, because we processTokenResponse even if we've got IOException